packages/caravan-health/src/waste.ts

```diff
@@ -125,6 +125,37 @@ export class WasteMetrics extends WalletMetrics {
     return weight * (feeRate - estimatedLongTermFeeRate) + costOfTx;
   }
 
+  /*
+    Name : calculateDustLimits
+    Definition : 
+      Dust outputs are uneconomical to spend because the fees to spend them are higher than 
+      the value of the output. So to avoid the situations for having dust outputs in the wallet,
+      we calculate the dust limits for the wallet.
+
+    Calculation :
+      lowerLimit - Below which the UTXO will actually behave as a dust output.
+      upperLimit - Above which the UTXO will be safe and economical to spend.
+      riskMultiplier - A factor that helps to determine the upper limit.
+
+      The average size of the transaction to move one UTXO value could be 250-400 vBytes.
+      So, for the lower limit we are taking 250 vBytes as the transaction weight by default.
+      If your wallet supports weight units, you can change the value accordingly.
+
+      lowerLimit = 250 * feeRate (sats/vByte)
+      upperLimit = lowerLimit * riskMultiplier
+  */
+  calculateDustLimits(
```

Comment:
This is pretty neat! 

I'm curious a little more about how this works. Why do we need a lower and upper bound? Isn't it binary where below a number we consider it dust and above we consider it spendable? 

With this calculation if we have feeRate of 1 sat/vbyte, the default we get for the dust limit is {lower: 250, upper: 500}. As far as I understand, we only really care about that 500 number (unfortunately it's also a little low from the common 546 default. I think it also depends on script type right? 

Maybe [this SO post](https://bitcoin.stackexchange.com/questions/86068/how-was-the-dust-limit-of-546-satoshis-was-chosen-why-not-550-satoshis) might be helpful

Reply:
On the lower and upper limit parts, So The size of the transaction to move one UTXO value could be in the range of 250-400 vB including the change and overhead.

So here 250 vB can be used as a lower limit and 400/500 vB as upper limit meaning that 500 will be the worst case weight (Will update the logic to consider script type and make it go as high as 546) But the weight will not be 546 in all cases. 

So,

We give a warning in the range of `lower_limit` to `upper_limit` that please do something about this UTXO or it will become dust very soon in future with a minor increase in fee rate.

Below `lower_limit` It's already a dust and we cannot do something about it at the moment. Maybe wait untill feerate goes down or else the UTXO is gone. 

Above `upper_limit` It's safe from becoming dust. No worries about this UTXO value sitting into the wallet.

---

Reply:
I see. Interesting! So that's more nuanced and detailed then what they do in bitcoin core. In core they [just do a] (https://github.com/bitcoin/bitcoin/blob/c536dfbcb00fb15963bf5d507b7017c241718bf6/src/policy/policy.cpp#L18-L34) calculation just based on script type: a utxo is dust if it costs more to send based on the size of the input. 

```
    // If you'd pay more in fees than the value of the output
    // to spend something, then we consider it dust.
    // A typical spendable non-segwit txout is 34 bytes big, and will
    // need a CTxIn of at least 148 bytes to spend:
    // so dust is a spendable txout less than
    // 182*dustRelayFee/1000 (in satoshis).
```

But your point is that there is some wiggle room based on the size of the transaction because of overhead. I believe a smaller tx needs each input to carry more of its own weight right? Essentially it's the percentage of overall size taken up by overhead and change. 

I think this is worth documenting thoroughly as it wasn't immediately obvious to me and not what bitcoin core does. Might also be worth adding a link to that relevant code from bitcoin core for the simple calculation for reference. And then add script type support and then this should be good.